tardis/transport/montecarlo/base.py

```diff
@@ -212,13 +213,14 @@ def run(
         # Condition for Checking if RPacket Tracking is enabled
         if self.montecarlo_configuration.ENABLE_RPACKET_TRACKING:
             transport_state.rpacket_tracker = rpacket_trackers
-
-        if self.transport_state.rpacket_tracker is not None:
             self.transport_state.rpacket_tracker_df = (
                 rpacket_trackers_to_dataframe(
                     self.transport_state.rpacket_tracker
                 )
             )
+        else:
+            transport_state.rpacket_tracker = rpacket_last_interaction_trackers
```

Comment:
What if both `rpacket_last_interaction_trackers` exists and `self.montecarlo_configuration.ENABLE_RPACKET_TRACKING` is true?

Reply:
That can't happen. Because while tracking the same if else condition is used.